storm-core/src/jvm/backtype/storm/scheduler/Cluster.java

```diff
@@ -430,12 +457,44 @@ public SupervisorDetails getSupervisorById(String nodeId) {
     }
 
     /**
+     * set assignments for cluster
+     */
+    public void setAssignments(Map<String, SchedulerAssignment> newAssignments) {
+        this.assignments = new HashMap<String, SchedulerAssignmentImpl>();
+        for (Map.Entry<String, SchedulerAssignment> entry : newAssignments.entrySet()) {
+            this.assignments.put(entry.getKey(), new SchedulerAssignmentImpl(entry.getValue().getTopologyId(), entry.getValue().getExecutorToSlot()));
```

Comment:
Why not add a copyOf method to SchedulerAssignmentImpl? 
`SchedulerAssignmentImpl.copyOf(entry.getValue())`

That should sort of clarify any confusion that you're just making `assignments` into a copy of `newAssignments`.